### PR TITLE
RC Link improvements

### PR DESCRIPTION
**Pull request by PurpleGuitar**
- Added regex parsing for parsing double-bracket links
- Removed unused override field
- Added tests

### CODE REVIEW

File: SRPTests/RCLinkTests.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
+using Markdig;
+using NUnit.Framework;
```
Comment (rbnswartz) [inline SRPTests/RCLinkTests.cs:2]: Out of curiosity why did you choose NUnit over MSTest?

File: SRPTests/RCLinkTests.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        private RCLinkOptions options;
+        private MarkdownPipeline pipeline;
+
+        [SetUp]
+        public void Setup()
+        {
+            this.options = new RCLinkOptions()
+            {
+                BaseUser = "WycliffeAssociates",
+                // TODO: this needs to be changed to a configuration value
```
Comment (rbnswartz) [inline SRPTests/RCLinkTests.cs:20]: This todo is technically false since this is in a test.

**Review by rbnswartz — APPROVED** [review]
Approved with suggestions

File: SRPTests/RCLinkTests.cs (hunk 1 above)
Comment (PurpleGuitar) [reply SRPTests/RCLinkTests.cs:2]: I seemed to remember we've used NUnit in the past.  I have no preference; happy to switch over the MSTest if you'd prefer.

File: SRPTests/RCLinkTests.cs (hunk 2 above)
Comment (PurpleGuitar) [reply SRPTests/RCLinkTests.cs]: You're right.  Fixed.

**Pull request merged**